### PR TITLE
Add support for concurrency mocking

### PR DESCRIPTION
**Pull request by ailtonvivaz**
This PR implements support for parsing async methods and closures  as discussed in #407

### CODE REVIEW

File: Generator/Source/CuckooGeneratorFramework/Templates/NopImplStubTemplate.swift [hunk 1]
```diff
@@ -31,6 +34,12 @@ extension Templates {
     {% endfor %}
 
     {% for method in container.methods %}
+    {% if debug %}
+    // {{method}}
+    {% endif %}
+    {% for attribute in method.attributes %}
```
Comment (MatyasKriz) [inline Generator/Source/CuckooGeneratorFramework/Templates/NopImplStubTemplate.swift:40]: What do these attributes do and how are they linked to async/await?

File: Tests/Swift/Source/TestedProtocol.swift [hunk 2]
```diff
@@ -17,7 +17,7 @@ protocol TestedProtocol {
     func noReturn()
 
     func count(characters: String) -> Int
-
+
```
Comment (MatyasKriz) [inline Tests/Swift/Source/TestedProtocol.swift:20]: Please remove this whitespace. 🙂

**Review by MatyasKriz — COMMENTED** [review]
What a superb pull request! Well done, @ailtonvivaz. I have a minor question, but after that we can merge this immediately. I think it warrants its own release as this is quite a good feature to have, we just weren't able to find the time, so thank you very much!

File: Generator/Source/CuckooGeneratorFramework/Templates/NopImplStubTemplate.swift (hunk 1 above)
Comment (ailtonvivaz) [reply Generator/Source/CuckooGeneratorFramework/Templates/NopImplStubTemplate.swift:40]: The concurrency in Swift is available just for iOS 13+, macOS 10.15+,watchOS 6.0+ and tvOS 13.0+.
As the library works with iOS 8+, macOS 10.9+ and tvOS 9+, some users could use the @available attribute to provide some methods with this capability.
And, as `NopImplStubTemplate.swift` hadn't this implementation yet, I've added.

File: Tests/Swift/Source/TestedProtocol.swift (hunk 2 above)
Comment (ailtonvivaz) [reply Tests/Swift/Source/TestedProtocol.swift]: Thanks. It was removed

File: Generator/Source/CuckooGeneratorFramework/Templates/NopImplStubTemplate.swift (hunk 1 above)
Comment (MatyasKriz) [reply Generator/Source/CuckooGeneratorFramework/Templates/NopImplStubTemplate.swift:40]: I see, thanks for explaining. 🙂

**Comment by MatyasKriz** [comment]
I've merged this into `master`, but I can't seem to be able to build it with the current values in the `@available` attribute, Xcode keeps suggesting to raise them. I'll update Xcode to the latest version and try again.

**Comment by ailtonvivaz** [comment]
> I've merged this into `master`, but I can't seem to be able to build it with the current values in the `@available` attribute, Xcode keeps suggesting to raise them. I'll update Xcode to the latest version and try again.

Maybe your Xcode version is below 13.2. Before this, concurrency was supported just for iOS15+.
https://developer.apple.com/documentation/xcode-release-notes/xcode-13_2-release-notes

Updating to the new Xcode, you'll get this feature.

**Pull request closed without merging**